src/redisjson.rs

```diff
@@ -43,28 +43,35 @@ pub struct RedisJSON {
 
 impl RedisJSON {
     pub fn from_str(data: &str) -> Result<Self, Error> {
-        eprintln!("Parsing JSON from input '{}'", data);
-
         // Parse the string of data into serde_json::Value.
         let v: Value = serde_json::from_str(data)?;
 
         Ok(Self { data: v })
     }
 
-    pub fn set_value(&mut self, data: &str) -> Result<(), Error> {
-        eprintln!("Parsing JSON from input '{}'", data);
-
+    pub fn set_value(&mut self, data: &str, path: &str) -> Result<(), Error> {
         // Parse the string of data into serde_json::Value.
-        let v: Value = serde_json::from_str(data)?;
+        let json: Value = serde_json::from_str(data)?;
 
+        let v = jsonpath_lib::replace_with(self.data.clone(), path, &mut |_v| {
```

Comment:
1. I suggest renaming `RedisJSON::data` to `RedisJSON::value`, since too many things are called `data`.
1. I don't see any reason to clone `self.data` if it's going to be replaced anyway.

I'd suggest the following instead:

```
    pub fn set_value(&mut self, data: &str, path: &str) -> Result<(), Error> {
        let new_value: Value = serde_json::from_str(data)?;

        let current_value = mem::replace(&mut self.value, Value::Null);
        let v = jsonpath_lib::replace_with(current_value, path, &mut |_v| {
            new_value.clone()
        })?;

        self.value = v;

        Ok(())
    }
```

Reply:
1. I'll do i on another PR
2. Done